packages/scm/src/browser/scmService.ts

```diff
@@ -0,0 +1,187 @@
+/********************************************************************************
+ * Copyright (C) 2018 Red Hat, Inc. and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+import {
+    InputValidator,
+    ScmInput,
+    ScmProvider,
+    ScmRepository,
+    ScmService
+} from '../common/scm';
+import { Disposable, Emitter, Event } from '@theia/core/lib/common';
+import { injectable } from 'inversify';
+
+@injectable()
+export class ScmServiceImpl implements ScmService {
```

Comment:
please follow project file naming conventions, i.e. `scm-service-impl.ts`

put classes to its own files, name files after classes, it makes easier to find them and does not produce duplicate file names in the file quick pick

Reply:
fixed